src/components/menu/menu-directive.ts

```diff
@@ -65,6 +70,60 @@ export class MdMenu {
     this._showClickCatcher = bool;
   }
 
+  /**
+   * This method is used by the menu trigger to focus the first item when the menu is opened
+   * by the ENTER key.
+   * TODO: internal
+   */
+  _focusFirstItem() { this.items.first.focus(); }
+
+  // TODO(kara): update this when (keydown.downArrow) testability is fixed
+  private _checkKey(event: KeyboardEvent): void {
+    if (event.keyCode === DOWN_ARROW) {
+      this._focusNextItem();
+    } else if (event.keyCode === UP_ARROW) {
+      this._focusPreviousItem();
+    } else if (event.keyCode === TAB) {
+      this._checkTabDirection(event.shiftKey);
+    }
+  }
+
+  private _checkTabDirection(shiftPressed: boolean): void {
+    shiftPressed ? this._updateFocusedItem(-1) : this._updateFocusedItem(1);
+  }
+
+  private _emitCloseEvent(): void {
+    this._focusedItem = 0;
+    this.close.emit(null);
+  }
+
+  private _checkMenuBounds(): void {
+    if (this._focusedItem === this.items.length || this._focusedItem < 0) {
+      this._emitCloseEvent();
```

Comment:
It does close the menu. If someone tries to tab or arrow past the menu items in either direction, the menu should close.  The trigger is subscribed to the "close" event. When it's emitted from the menu, the trigger will close it.